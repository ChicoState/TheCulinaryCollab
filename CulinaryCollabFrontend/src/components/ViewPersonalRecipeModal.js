import React from 'react';
import './ViewRecipeModal.css';

const PersonalRecipeViewModal = ({ isOpen, onClose, recipe, onEdit }) => {
	if (!recipe) return null;
	const handleEditClick = () => {
		onEdit();
		onClose();
	}
	return (
		<div className={`modal ${isOpen ? 'open' : ''}`}>
		<div className="modal-content">
		<button className="edit-button" onClick={handleEditClick}>Edit</button>
		<h2>{recipe.name}</h2>
		<p><strong>Timing:</strong> {recipe.timing}</p>
		<p><strong>Taste:</strong> {recipe.taste}</p>
		<p><strong>Ingredients:</strong></p>
		<ul>
		{recipe.ingredients.map((ingredient, index) => (
			<li key={index}>
			{ingredient.ingredient} ({ingredient.amount} {ingredient.unit})
			</li>
		))}
		</ul>
		<p><strong>Preparation:</strong> {recipe.preparation}</p>
		<button onClick={onClose}>Close</button>
		</div>
		</div>
	);
};

export default PersonalRecipeViewModal;

