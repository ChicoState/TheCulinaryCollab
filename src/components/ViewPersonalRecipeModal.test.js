import React from 'react';
import { render, fireEvent, waitFor, findByText } from '@testing-library/react';
import '@testing-library/jest-dom';
import ViewPersonalRecipeModal from './ViewPersonalRecipeModal';

const mockRecipe = {
  id: 1,
  name: 'Rusty Nail',
  timing: 'After dinner',
  taste: 'Boozy',
  ingredients: [
    { amount: 4.5, ingredient: 'Scotch Whisky', unit: 'cl' },
    { amount: 2.5, ingredient: 'Drambuie', unit: 'cl' },
  ],
  preparation: 'Stirred',
  cost: undefined,
  timeToMake: undefined,
};

describe('ViewPersonalRecipeModal', () => {
  test('renders the component with a recipe', async () => { 
    const { container, getByText } = render(
      <ViewPersonalRecipeModal isOpen={true} onClose={() => {}} recipe={mockRecipe} onEdit={() => {}} onDelete={() => {}} />
    );
  
    await waitFor(() => { 
      expect(container).toMatchSnapshot();
    });
    const timingElement = await findByText('Timing: After dinner');
      expect(timingElement).toBeInTheDocument();
      expect(getByText('Rusty Nail')).toBeInTheDocument();
      expect(getByText('Timing: After dinner')).toBeInTheDocument();
      expect(getByText('Taste: Boozy')).toBeInTheDocument();
      expect(getByText('Scotch Whisky (4.5 cl)')).toBeInTheDocument();
      expect(getByText('Drambuie (2.5 cl)')).toBeInTheDocument();
      expect(getByText('Preparation: Stirred')).toBeInTheDocument();
      expect(getByText('Cost:')).not.toBeInTheDocument();
      expect(getByText('Time to Make:')).not.toBeInTheDocument();

  });

  test('calls onEdit and onClose when Edit button is clicked', () => {
    const onEditMock = jest.fn();
    const onCloseMock = jest.fn();

    const { getByText } = render(
      <ViewPersonalRecipeModal isOpen={true} onClose={onCloseMock} recipe={mockRecipe} onEdit={onEditMock} onDelete={() => {}} />
    );

    const editButton = getByText('Edit');
    fireEvent.click(editButton);

    expect(onEditMock).toHaveBeenCalledWith(1); // Assuming the recipe id is 1
    expect(onCloseMock).toHaveBeenCalled();
  });

  test('calls onDelete and onClose when Delete button is clicked', () => {
    const onDeleteMock = jest.fn();
    const onCloseMock = jest.fn();

    const { getByText } = render(
      <ViewPersonalRecipeModal isOpen={true} onClose={onCloseMock} recipe={mockRecipe} onEdit={() => {}} onDelete={onDeleteMock} />
    );

    const deleteButton = getByText('Delete');
    fireEvent.click(deleteButton);

    expect(onDeleteMock).toHaveBeenCalledWith(1); // Assuming the recipe id is 1
    expect(onCloseMock).toHaveBeenCalled();
  });

  // Add more tests as needed
});

