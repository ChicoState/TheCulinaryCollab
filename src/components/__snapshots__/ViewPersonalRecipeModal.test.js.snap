// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewPersonalRecipeModal renders the component with a recipe 1`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 2`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 3`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 4`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 5`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 6`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 7`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 8`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 9`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 10`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 11`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 12`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 13`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 14`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 15`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 16`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 17`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 18`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ViewPersonalRecipeModal renders the component with a recipe 19`] = `
<div>
  <div
    class="modal open"
  >
    <div
      class="modal-content"
    >
      <button
        class="edit-button"
      >
        Edit
      </button>
      <div
        class="average-rating-section"
      >
        <p>
          Average Rating: 
          <div
            class="star-rating"
          >
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
            <span
              class="star "
            >
              ★
            </span>
          </div>
          <span
            class="ratings-count"
          >
             (
            0
             Ratings)
          </span>
        </p>
        <div
          class="comments-toggle"
        >
          <button>
            Show Comments (
            0
            )
          </button>
        </div>
      </div>
      <h2>
        Rusty Nail
      </h2>
      <p>
        <strong>
          Timing:
        </strong>
         
        After dinner
      </p>
      <p>
        <strong>
          Taste:
        </strong>
         
        Boozy
      </p>
      <p>
        <strong>
          Ingredients:
        </strong>
      </p>
      <ul>
        <li>
          Scotch Whisky
           (
          4.5
           
          cl
          )
        </li>
        <li>
          Drambuie
           (
          2.5
           
          cl
          )
        </li>
      </ul>
      <p>
        <strong>
          Preparation:
        </strong>
         
        Stirred
      </p>
      <button>
        Delete
      </button>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;
